src/data/cassandra/impl/Cluster.hpp

```diff
@@ -31,10 +31,29 @@
 #include <string>
 #include <string_view>
 #include <thread>
+#include <utility>
 #include <variant>
 
 namespace data::cassandra::impl {
 
+namespace {
+
+enum class Provider { Cassandra, Keyspace };
+
+inline std::string
+toString(Provider provider)
```

Comment:
You will have a function providerFromString instead, and this function could be removed